tokio-codec/tests/codecs.rs

```diff
@@ -55,6 +55,48 @@ fn lines_decoder() {
     assert_eq!(None, codec.decode_eof(buf).unwrap());
 }
 
+#[test]
+fn lines_decoder_max_length() {
+    let mut codec = LinesCodec::new();
+    codec.set_decode_max_line_length(5);
+    let buf = &mut BytesMut::new();
+    buf.reserve(200);
+    buf.put("line 1 is too long\nline 2\r\nline 3\n\r\n\r");
+    assert_eq!("line 1", codec.decode(buf).unwrap().unwrap());
```

Comment:
Hmmm, this looks weird to me.  These fragments are all six characters long, including whitespace.

Reply:
Whoops, I can't count. I'll fix the test (and the off-by-one error it failed to catch :P)

Reply:
Not that I'm dying for this or anything, but, have you had any time to look at this?  I can try picking it up if you're on higher-priority stuff.